include/open62541/server.h

```diff
@@ -1886,6 +1891,57 @@ typedef struct {
 UA_ServerStatistics UA_EXPORT
 UA_Server_getStatistics(UA_Server *server);
 
+/**
+  * Reverse Connect
+  * ---------------
+  *
+  * The reverse connect feature of OPC UA permits the server instead of the client to establish
+  * the connection. The client must expose the listening port so the server is able to reach it.
+  */
+typedef enum {
+    UA_REVERSECONNECTSTATE_DISCONNECTED = 0,
+    UA_REVERSECONNECTSTATE_CONNECTING,
+    UA_REVERSECONNECTSTATE_CONNECTED,
+} UA_ReverseConnectState;
+
+/**
+ * The reverse connect state change callback is called whenever the state of a reverse connect
+ * is modified by a connection attempt, a successful connection or a connection loss.
+ *
+ * Each attempt to establish a connection will lead to a call with UA_REVERSECONNECTSTATE_CONNECTING
+ * followed by either UA_REVERSECONNECTSTATE_CONNECTED if a secure channel was successfully established
+ * or UA_REVERSECONNECTSTATE_DISCONNECTED if the connection attempt was not successful.
+ * A connection break leads to a call with UA_REVERSECONNECTSTATE_DISCONNECTED followed by
+ * UA_REVERSECONNECTSTATE_CONNECTING when the next reconnect attempt is made.
```

Comment:
From my point of view, the handle refers to a reverse connect configuration and not to a connection.

I guess most users will use reverse connect to get around strange firewall configurations and just want to enable a client to talk to the server. Not doing reconnect attempts automatically on connection  loss would mean that the user has to implement the state callback and keep tabs on the association between handle and remote client URL to add a new reverse connect once the connection has been lost.

Doing reconnects by default would cater to the "lazy" user and anyone who would want control could implement the state change callback and remove the reverse connect configuration from inside the callback by using the handle.

Reply:
That works as well.
The importance is to have no ambiguity in the documentation.
So that people can rely on a behavior.

That means a connection is never ‚disconnected‘.
If a reconnect attempt is made immediately.
So the. We only have ‚connecting‘ and ‚connected‘ as status.

Reply:
@jpfr The attempt is made periodically by the repeated callback. Depending on when the connection breaks, this could be right away or after a few seconds.
What about a WaitingForReconnect state?

Reply:
@jpfr Sounds good, I have updated the PR accordingly

Reply:
Great!
The API is good for me.
Just trigger us when we should take a look before merging.